autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -9,14 +9,63 @@
 import scipy.sparse
 
 import sklearn.utils
-from sklearn import preprocessing
+
 from sklearn.base import BaseEstimator
 from sklearn.compose import ColumnTransformer
 from sklearn.exceptions import NotFittedError
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler, OrdinalEncoder
 
 from autoPyTorch.data.base_feature_validator import BaseFeatureValidator, SUPPORTED_FEAT_TYPES
 
 
+def _create_column_transformer(preprocessors: typing.Dict, numerical_columns, categorical_columns):
+    numerical_pipeline = 'drop'
+    categorical_pipeline = 'drop'
+    if len(numerical_columns) > 0:
+        numerical_pipeline = make_pipeline(*preprocessors['numerical'])
+    if len(categorical_columns) > 0:
+        categorical_pipeline = make_pipeline(*preprocessors['categorical'])
+
+    return ColumnTransformer([
+        ('categorical_pipeline', categorical_pipeline, categorical_columns),
+        ('numerical_pipeline', numerical_pipeline, numerical_columns)],
+        remainder='passthrough'
+    )
+
+
+def get_tabular_preprocessors():
+    preprocessors = dict()
+    preprocessors['numerical'] = list()
+    preprocessors['categorical'] = list()
+
+    preprocessors['categorical'].append(OneHotEncoder(
+        categories='auto',
+        sparse=False,
+        handle_unknown='ignore'))
+    preprocessors['categorical'].append(SimpleImputer(strategy='constant',
+                  # Train data is numpy
+                  # as of this point, where
+                  # Ordinal Encoding is using
+                  # for categorical. Only
+                  # Numbers are allowed
+                  # fill_value='!missing!',
+                  fill_value=-1,
+                  copy=False))
+
+    preprocessors['categorical'].append(OrdinalEncoder(
```

Comment:
The ordering seems weird, is it reversed?
OneHotEncoder should only be called in the end.
Additionally, why do we need the ordinal encoder? Does one hot encoder not handle the case of let us say 4 unique values str and -1?  The ordering should be something like Simple imputer, Onehot encoder

Reply:
I have fixed this now.